drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c

```diff
@@ -561,3 +563,7 @@ DEVICE_DEFINE(wlan0, /* name - token */
 	      CONFIG_WIFI_INIT_PRIORITY, /* prio */
 	      NULL); /* api */
 #endif /* CONFIG_WPA_SUPP */
+
+#ifdef CONFIG_L2_WIFI_CONNECTIVITY
+CONN_MGR_BIND_CONN(wlan0, L2_CONN_WLAN0);
```

Comment:
I am bit skeptical about this, the interfaces to be used by connection should be in a single place, declaring `L2_CONN_WLAN0` in conn_mgr and `wlan0` in driver would be bit confusing. Can't we do this in `l2_wifi_conn.c` itself? We are anyways using `L2_CONN_WLAN0`, there is a implicit assumption of 121 mapping.

Reply:
I would rename `L2_CONN_WLAN0` to `L2_CONN_WLAN` -- In a hypothetical scenario with multiple WLAN's, they should ideally share the connectivity implementation

As for the location of `CONN_MGR_BIND_CONN` it unfortunately has to be in the same file as the `DEVICE_DEFINE` call since `DEVICE_DEFINE` creates a static struct internally

Reply:
Having same connectivity binding across multiple interfaces, wouldn't that overwrite the net_iface that we are storing in a global variable, when same init call gets invoked for all wifi interfaces?

Reply:
> In a hypothetical scenario with multiple WLAN's

That's 1/2 NCS releases away :)

Reply:
> wouldn't that overwrite the net_iface that we are storing in a global variable, when same init call gets invoked for all wifi interfaces?

Yes, that's correct. You would have to stop using the global variable and start using the context pointers. And you'd need to instance timeout workqueue items for each iface with a binding as well.